pkg/inject/sidecar.go

```diff
@@ -80,6 +73,15 @@ func Needed(dep *appsv1.Deployment, ns *corev1.Namespace) bool {
 	return !HasJaegerAgent(dep)
 }
 
+func setInjectedInstanceLabel(target *appsv1.Deployment, jaeger *v1.Jaeger) {
```

Comment:
In new code, we are trying to avoid changing the parameters we receive, returning new values instead. Like:

```go
func setInjectedInstanceLabel(target *appsv1.Deployment, jaeger *v1.Jaeger) *appsv1.Deployment {
  ret := *target
  ...
  return &ret
}
```

In this particular case, I would actually not split it in a separate function, as it this is very simple and, apparently, needed only in the `Sidecar` function.

Reply:
I don't remember why I split this.. I think it was due some codeclimate issues. I'll check but seems to not be necessary